go/libraries/doltcore/table/editor/sessioned_table_editor.go

```diff
@@ -419,7 +447,16 @@ func (ste *sessionedTableEditor) validateForInsert(ctx context.Context, taggedVa
 		if !ok {
 			return fmt.Errorf("unable to get table editor as `%s` is missing", foreignKey.ReferencedTableName)
 		}
-		exists, err := ContainsIndexedKey(ctx, referencingSte.tableEditor, indexKey, foreignKey.ReferencedTableIndex)
+
+		tblName := referencingSte.tableEditor.Name()
+		cacheKey := tblName + "->" + foreignKey.ReferencedTableIndex
+		idxSch, ok := ste.indexSchemaCache[cacheKey]
+		if !ok {
+			idxSch = referencingSte.tableEditor.Schema().Indexes().GetByName(foreignKey.ReferencedTableIndex).Schema()
+			ste.indexSchemaCache[cacheKey] = idxSch
+		}
+
+		exists, err := ContainsIndexedKey(ctx, referencingSte.tableEditor, indexKey, foreignKey.ReferencedTableIndex, idxSch)
```

Comment:
could we factor a common `func (ste *sessionedTableEditor) getIndex(tblName, idxName string) ...`?